src/test/groovy/org/gradle/android/TaskAvoidanceTest.groovy

```diff
@@ -0,0 +1,55 @@
+package org.gradle.android
+
+import org.gradle.util.GradleVersion
+import org.gradle.util.VersionNumber
+import spock.lang.Unroll
+
+class TaskAvoidanceTest extends AbstractTest {
+    @Unroll
+    def "tasks are avoided with #gradleVersion and Android plugin #androidVersion"() {
+        assert gradleVersion instanceof GradleVersion
+        assert androidVersion instanceof VersionNumber
+
+        println "> Using Android plugin $androidVersion"
+        println "> Running with $gradleVersion"
+
+        SimpleAndroidApp.builder(temporaryFolder.root, cacheDir)
+                .withAndroidVersion(androidVersion)
+        .withKotlinDisabled()
+                .build()
+                .writeProject()
+
+        def originalSettings = file('build.gradle').text
+        file('build.gradle').text = """
+            ${originalSettings}
+
+            allprojects {
+              tasks.configureEach {
+                println "configuring \$it"
+              }
+            }
+        """
```

Comment:
I believe this can be simplified with `file('build.gradle') << " .... "

Reply:
nice groovy magic